File: C++/Fundamentals/Testing 1-2-3.cpp
#include <iostream>
#include <vector>
#include <string>

std::vector<std::string> number(std::vector<std::string> const &lines)
{
    if (lines.empty())
        return {};

    std::vector<std::string> num(lines.cbegin(), lines.cend());
    for (size_t i{}; i < num.size(); i++)
        num.at(i).insert(0, std::to_string(i + 1) + ": ");
    return num;
}

int main()
{
    std::vector<std::string> v{"a", "b", "c"};
    for (auto const &el : number(v))
        std::cout << el << '\n';

    return EXIT_SUCCESS;
}

File: Python/Fundamentals/Testing 1-2-3.py
def number(lines: list) -> list:
    for i, el in enumerate(lines):
        lines[i] = str(i + 1) + ': ' + el
    return lines


def number_smart(lines: list) -> list:
    return ['%d: %s' % v for v in enumerate(lines, 1)]


print(number(['a', 'b', 'c']))
print(number_smart(['a', 'b', 'c']))
